js/btcmarkets.js

```diff
@@ -631,111 +675,99 @@ module.exports = class btcmarkets extends Exchange {
         return result;
     }
 
+    parseOrderStatus (status) {
+        const statuses = {
+            'Accepted': 'open',
+            'Placed': 'open',
+            'Partially Matched': 'open',
+            'Fully Matched': 'closed',
+            'Cancelled': 'canceled',
+            'Partially Cancelled': 'canceled',
+            'Failed': 'rejected',
+        };
+        return this.safeString (statuses, status, status);
+    }
+
     parseOrder (order, market = undefined) {
-        const multiplier = 100000000;
-        const side = (order['orderSide'] === 'Bid') ? 'buy' : 'sell';
-        const type = (order['ordertype'] === 'Limit') ? 'limit' : 'market';
-        const timestamp = this.safeInteger (order, 'creationTime');
+        const timestamp = this.parse8601 (this.safeString (order, 'creationTime'));
+        const marketId = this.safeString (order, 'marketId');
         if (market === undefined) {
```

Comment:
Instead of lines 694-703
We should do the following:

```
        const marketId = this.safeString (order, 'marketId');
        if (marketId !== undefined) {
            if (marketId in this.markets_by_id) {
                market = this.markets_by_id[marketId];
            } else {
                const [ quoteId, baseId ] = marketId.split ('-');
                const base = this.safeCurrencyCode (baseId);
                const quote = this.safeCurrencyCode (quoteId);
                symbol = base + '/' + quote;
            }
        }
        if ((symbol === undefined) && (market !== undefined)) {
            symbol = market['symbol'];
        }
```

↑ It's the same chunk of logic everywhere throughout the file.

Reply:
It is in line with the other exchanges, and we will detach it into a standalone base method later

Reply:
Ok, have created a method.  Wasn't sure if there was a standard name